README.md
# plugin.video.sosac.ph

[![Codacy Badge](https://api.codacy.com/project/badge/Grade/d6eb929d7cbb4c2681bffa6a43dcd9e3)](https://www.codacy.com/app/zbyna/plugin-video-sosac-ph?utm_source=github.com&utm_medium=referral&utm_content=zbyna/plugin.video.sosac.ph&utm_campaign=badger)
[![Code Climate](https://codeclimate.com/github/zbyna/plugin.video.sosac.ph/badges/gpa.svg)](https://codeclimate.com/github/zbyna/plugin.video.sosac.ph)

[![Issue Count](https://codeclimate.com/github/zbyna/plugin.video.sosac.ph/badges/issue_count.svg)](https://codeclimate.com/github/zbyna/plugin.video.sosac.ph)

[![Code Health](https://landscape.io/github/zbyna/plugin.video.sosac.ph/master/landscape.svg?style=flat)](https://landscape.io/github/zbyna/plugin.video.sosac.ph/master)

Tato verze je založena na verzi 1.2.11 https://github.com/kodi-czsk/plugin.video.sosac.ph (fork)

Děkuji autorům.

Pro správnou funkci doplňku je třeba **nejprve** nainstalovat také **aktuální verzi stream.resolveru** z https://github.com/zbyna/script.module.stream.resolver/releases

# Rozšíření a úpravy:

![](http://i.imgur.com/f0VVTHB.png)
- při **přehrávání z media library** přidává:
  1. podporu pro **automatické nastavení statusu zhlédnuto** 
     při min 90 % přehrání položky  
  2. **osd menu s posterem** (filmy), **náhledem** (seriály), **dějem
     a obsazením**, vyžaduje stream.resolver min 1.6.435 a podporu skinu 
     
- přidáno **automatické obnovení přehrávání při opětovném spuštění videa z media library**.  
  Bod obnovení se vytvoří pouze když je přehráno minimálně 5 a max 90 % videa.
  Při znovu přehrání videa je na výběr zda obnovit přehrávání nebo přehrát od začátků.
  Po vybrání jedné z voleb se bod obnovení vymaže.  
  
- přidán **jednoduchý manažer odběru pro seriály** 

- přidána položka **"Filmy dle roku"**

- přidáno **"Žebříčky csfd"** (nejlepší filmy, nejlepší seriály, celkově
  a podle žánru) 
  
- přidáno **"Ocenění z csfd"** (Oskary, Zlatá palma) 

- přidáno kont. menu **"Přidej vše do knihovny"** pro **žánr** ve "Filmy podle žánrů" a 
  **rok** ve "Filmy podle roku" 
  
- přidáno kont. menu **"Přihlas k odběru"** pro položky ve správci odběrů a 
  **"Odhlásit všechny odběry"** pro root položku "Správce odběrů" 
  
- možnost **nepřihlašovat k odběru při přidání všech** seriálů do knihovny

- přidáno info o dabingu do názvu filmu či seriálů 

- přidáno  stream info - video, audio,titulky 
  - nutná podpora ikonek ve skinu (vyzkoušen AEON MQ.7)  a stream resolver alespoň 2.0

 ![](http://i.imgur.com/hO4Xg3k.jpg)

- přidána možnost uživatelského scanu odběrů:
  - **Nastavení/Služba knihovny/Zkontroluj odběry**
  
  - **kontextové menu Správce odběru/Zkontroluj odběry** 

- pridáno **implementace memory cache** pro časově a početně náročné operace 

- přidáno **implementace multithread** zpracování u IO operací

- přidáno nastavení doplňku 
    - **karta Obecné - vynutit anglické názvy**,
      1. **Třídit podle - czech, english, os**
          - pokud v operačním systému neexistuje zvolená lokalizace tak se vybere defaultní, 
je to problém třeba v LibreElec nebo OpenElec, kde skutečně lokalizace cs_CZ.utf8 není k dispozici, ale blíská se na lepší časy:  https://forum.libreelec.tv/thread-7356.html
          - podle normy je české třídění nejprve abeceda až po ní písmena, pro mě překvapení 🙂 , windows to vůbec neřeší, ale v Linuxu pokud teda je lokalizace nainstalovaná, vše funguje podle normy

      2. **Zobrazit 'CH' ve Filmy a Seriály**
		    - zařadí 'ch' do seznamu Filmy a Seriály
        
    - **karta Cache - vyprádnit cache**, 
    
    - **karta Služba knihovny - odběr a přidání do knihovny**
      - pro nastavení možnosti současného provedení těchto položek 
                              
- přidána možnost **stahování na SMB zdroje**, vyžaduje  stream.resolver min. 1.6.447 

- přídán **překlad dosud nelokalizovaných položek addonu** - do angličtiny, slovenštiny, češtiny

- přidána **podpora addonu Buggalo** pro volitelné odeslání chybového hlášení, po odsouhlasení uživatelem, na web:     sosac.comli.com 
- přidána možnost **nastavení knihovny sosáče na libovolný zdroj přímo z Kodi**,
  (po obnovení defaultních hodnot (special:// ...) je  třeba restartovat sosáč)
  
- opraven **download souborů s titulky**, vyžaduje stream.resolver min. 1.6.435 

- opraveno **získávání titulků při přehrávání z media library** na pomalejším internetovém
  připojení, vyžaduje stream.resolver min. 1.6.447 

- opraveno WARNING: XFILE::CFileFactory::CreateLoader -,  vyžaduje stream.resolver min 1.6.435  

- opraveno načítání titulků pro Krypton při přehrávání mimo Kodi media library  
  (vyžaduje stream resolver min. 1.6.448)

  


changelog.txt
[B]2.0.200:[/B] 30. 5. 2017
- návrat Prihlas k odběru do kontextového menu Seriály (zbyna)
- přidáno Přidej vše do knihovny pro příslušné položky csfd seznamů (zbyna)
- opraveno při vymazání cache se nemaže Seznam odběrů (zbyna)
- přidán update strm souboru při opakovaném scanu pro filmy přidané do knihovny
  (zbyna)
- přidána experimentální podpora pro přehrávání položek TEST HTML5 (zbyna)
[B]2.0.100:[/B] 14. 5. 2017
- celkové výrazné zrychlení pluginu:
  přepsání subscribe manageru a zapnutí další úrovně cache (zbyna)
  zapnutí cache pro parsování u csfd seznamů  (zbyna)
- přidáno info o dabingu do názvu filmu či seriálů (zbyna)
- přidáno  stream info - video, audio,titulky (zbyna)
  nutná podpora ikonek ve skinu (vyzkoušen AEON MQ.7)
  a stream resolver alespoň 2.0
- přidáno popis k položkám Seriálů (zbyna)
- přidáno možnost uživatelského scanu odběrů:
  Nastavení/Služba knihovny/Zkontroluj odběry
  kontextové menu Správce odběru/Zkontroluj odběry (zbyna)
- přidáno automatické uzavření nastavení pluginu po vymazání cache (zbyna)
- přidány další volby v Nastavení/Obecné/Vynutit anglické názvy:
  Třídit podle - czech, english, os
  zobrazit 'CH' ve Filmy a Seriály  (zbyna)
- opraveno čitelnost v Nastavení/Služba knihovny (zbyna)
- opraveno odložení prvního startu Subscription service
- opraveno hláška při vymazání cache (zbyna)
- opraveno možnost přidat neexistující položku do knihovny 
  v csfd seznamech (zbyna)
[B]2.0.000:[/B] 19. 3. 2017
- přidáno žebříčky csfd (nejlepší filmy, nejlepší seriály, celkově
  a podle žánru) (zbyna)
- přidáno ocenění z csfd (Oskary, Zlatá palma) (zbyna)
- pridáno implementace memory cache pro časově a početně náročné operace (zbyna)
  možnost vypnout memory cache v nastavení doplňku prozatím není funkční
  (čeká se až se tento patch: 
  https://github.com/marcelveldt/script.module.simplecache/pull/2
  dostane do repozitáře kodi:
  https://github.com/xbmc/repo-scripts/tree/jarvis/script.module.simplecache)
- přidáno implementace multithread zpracování u IO operací
- přidáno nastavení doplňku karta Obecné - vynutit anglické názvy
- přidáno nastavení doplňku karta Cache - vyprádnit cache
- přidáno nastavení doplňku karta Služba knihovny - odběr a přidání do knihovny,
  pro nastavení možnosti současného provedení těchto položek (zbyna)
- opraveno načítání Nově přidaných seriálů (zbyna)
[B]1.2.118:[/B] 1. 1. 2017
- opravena funkčnost menu "Přidej vše do knihovny" pro "Filmy", "Seriály" a 
  "Nově přidané filmy" (zbyna)
- přidána položka "Filmy dle roku" (zbyna)
- přidáno kont. menu "Přidej vše do knihovny" pro žánr ve "Filmy podle žánrů" a 
  rok ve "Filmy podle roku" (zbyna)
- přidáno kont. menu "Přihlas k odběru" pro položky ve správci odběrů a 
  "Odhlásit všechny odběry" pro root položku "Správce odběrů" (zbyna)
- prohození položek v dialogu pro obnovení přehrávání (zbyna)
- možnost nepřihlašovat k odběru při přidání všech seriálů do knihovny (zbyna)
[B]1.2.117:[/B] 19. 12. 2016
- automatické obnovení přehrávání při opětovném spuštění videa
  z knihovny (zbyna)
- přídán překlad dosud nelokalizovaných položek addonu - do angličtiny,  
  slovenštiny, češtiny (zbyna)
- přidán jednoduchy manažer odběru pro seriály (zbyna)
- přidána možnost stahování na SMB zdroje, vyžaduje 
  stream.resolver 1.6.446 (zbyna)
- opraveno přehrávání z knihovny pro strm soubory umístěné na SMB zdroji
- opraveno získávání titulků při přehrávání z knihovny na pomalejším internetovém
  připojení, vyžaduje stream.resolver 1.6.446 (zbyna)
- opraveno  nastavení knihovny sosáče na libovolný zdroj přímo z Kodi,
  po obnovení defaultních hodnot (special:// ...) je 
  třeba restartovat sosáč (zbyna) 
[B]1.2.116:[/B] 31. 10. 2016
- opraveno exceptions.ImportError
  viz. error id=15  na sosac.comli.com (zbyna)
- opraveno zatuhnutí při pře přehrávání
  nenascrapovaných strm souborů (zbyna)
- opraveno korektní ukončení přehrávaní položky
  spouštějící migraci strm souborů (zbyna)
- snížení spotřeby paměti při migraci velkého
  počtu strm souborů (zbyna)
- zrychlení nastavení statusu shlédnuto pro
  rozsáhlé media knihovny (zbyna)
[B]1.2.115:[/B] 31. 10. 2016
- při přehrávání z media library přidává:
  1. podporu pro automatické nastavení statusu shlédnuto 
     při min 90 % přehrání položky  (zbyna)
  2. osd menu s posterem (filmy), náhledem (seriály), dějem
     a obsazením, vyžaduje stream.resolver 1.6.435  (zbyna) 
- přidána podpora addonu Buggalo pro volitelné odeslání
  chybového hlášení, po odsouhlasení uživatelem, 
  na web: sosac.comli.com (zbyna)
- opraveno download souborů s titulky,
  vyžaduje stream.resolver 1.6.435  (zbyna)
- opraveno WARNING: XFILE::CFileFactory::CreateLoader -,
  vyžaduje stream.resolver 1.6.435  (zbyna)
[B]1.2.11:[/B] 15. 10. 2016
- prepsan PRINT na UTIL.Info/Debug/Error (ExSport)
- STRING konverze, redukce SPAMu (ExSport)
- zobrazena kvalita u filmu (ExSport)
- odstranen problem s ruznymi typy objektu (ExSport)
[B]1.2.10:[/B] 27. 08. 2016
- fix osd diacritic when played from Media Library (zbyna)
[B]1.2.9:[/B] 26. 08. 2016
- sosac:Fix subscription update (zbyna)
[B]1.2.8:[/B] 12. 08. 2016
- zobrazeni CZ nazvu misto EN pri nastaveni Kodi-SK (ExSport)
[B]1.2.7:[/B] 02. 08. 2016
- zrychleni importu "Movies" do knihovny (ExSport)
[B]1.2.6:[/B] 31. 07. 2016
- subscription improvements (Libor Zoubek)
- localized service log messages to english (Libor Zoubek)
- fix #47,get rid of xbmc import (Libor Zoubek)
- filmy podla zanru (Michal Paulovic)
- opravy a pridane fce (ExSport)
- add Slovak translation (jose1711)
- code Climate: Check the code using pep8 (Lubo)
- reformat the code (Lubo)
- fix module level imports not being placed at top of file Lubo)
- README: Add Code Climate badge
[B]1.2.5:[/B] 26. 03. 2016
- Oprava #41 Kdy plugin blokoval kodi v ukonční (Kuci)
[B]1.2.4:[/B] 28. 02. 2016
- Opravené vyhľadanie na systémoch s nainštalovanou knižnicou lxml
[B]1.2.3:[/B] 20. 10. 2015
- opraveno stahování videí
[B]1.2.2:[/B] 06. 07. 2015
- přidána možnost hledání filmů

service.py
# -*- coding: UTF-8 -*-
# /*
#  *      Copyright (C) 2015 BBaronSVK
#  *
#  *
#  *  This Program is free software; you can redistribute it and/or modify
#  *  it under the terms of the GNU General Public License as published by
#  *  the Free Software Foundation; either version 2, or (at your option)
#  *  any later version.
#  *
#  *  This Program is distributed in the hope that it will be useful,
#  *  but WITHOUT ANY WARRANTY; without even the implied warranty of
#  *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
#  *  GNU General Public License for more details.
#  *
#  *  You should have received a copy of the GNU General Public License
#  *  along with this program; see the file COPYING.  If not, write to
#  *  the Free Software Foundation, 675 Mass Ave, Cambridge, MA 02139, USA.
#  *  http://www.gnu.org/copyleft/gpl.html
#  *
#  */

import xbmcaddon
from resources.lib.sosac import SosacContentProvider
from resources.lib.sutils import XBMCSosac

__scriptid__ = 'plugin.video.sosac.ph'
__scriptname__ = 'sosac.ph'
__addon__ = xbmcaddon.Addon(id=__scriptid__)
__language__ = __addon__.getLocalizedString
__set__ = __addon__.getSetting

settings = {'downloads': __set__('downloads'),
            'quality': __set__('quality'),
            'subs': __set__('subs') == 'true',
            'add_subscribe': __set__('add_subscribe')}

reverse_eps = __set__('order-episodes') == '0'

XBMCSosac(SosacContentProvider(reverse_eps=reverse_eps), settings, __addon__).service()


resources/lib/csfd.py
# -*- coding: UTF-8 -*-
from translatedStrings import *

csfd = {'level_0':
        [
            {
                'name': CSFD_LADDERS,
                'url': CSFD_BASE + 'zebricky/level_1',
                'type': 'DIR',
                'level_1': [
                    {
                        'name': BEST_MOVIES,
                        'url': CSFD_BASE + ZEBRICKY_FILMY_NEJ,
                        'type': 'DIR',
                        'menu':
                        {
                            "[B][COLOR red]" + ADD_ALL_TO_LIBRARY + "[/COLOR][/B]":
                            {
                                'action': 'add-all-to-library',
                                'title': BEST_MOVIES,
                                'url': CSFD_BASE + ZEBRICKY_FILMY_NEJ
                            }
                        }
                    },
                    {
                        'name': BEST_MOVIES_BY_GENRE,
                        'url': CSFD_BASE + ZEBRICKY_FILMY_SPEC,
                        'type': 'DIR'
                    },
                    {
                        'name': BEST_TV_SHOWS,
                        'url': CSFD_BASE + ZEBRICKY_TVSHOW_NEJ,
                        'type': 'DIR',
                        'menu':
                        {
                            "[B][COLOR red]" + ADD_ALL_TO_LIBRARY + "[/COLOR][/B]":
                            {
                                'action': 'add-all-to-library',
                                'title': BEST_TV_SHOWS,
                                'url': CSFD_BASE + ZEBRICKY_TVSHOW_NEJ
                            }
                        }
                    },
                    {
                        'name': BEST_TV_SHOWS_BY_GENRE,
                        'url': CSFD_BASE + ZEBRICKY_TVSHOW_SPEC,
                        'type': 'DIR'
                    }
                ]
            },
            {
                'name': CSFD_AWARDS,
                'url': CSFD_BASE + 'oceneni/level_1',
                'type': 'DIR',
                'level_1': [
                    {
                        'name': OSCARS,
                        'url': CSFD_BASE + OCENENI_OSCAR,
                        'type': 'DIR'},
                    {
                        'name': GOLDEN_PALM,
                        'url': CSFD_BASE + OCENENI_ZLATA_PALMA,
                        'type': 'DIR'
                    },

                ]
            }
        ]
        }


resources/lib/myPlayer.py
# -*- coding: UTF-8 -*-
import xbmc
import json
import time
from datetime import datetime, timedelta
import _strptime
import buggalo

# import pydevd
# pydevd.settrace(stdoutToServer=True,
#           stderrToServer=True,trace_only_current_thread=False)


class MyPlayer(xbmc.Player):

    def __init__(self, itemType=None, itemDBID=None, slovnik=None, titulky=None):
        try:
            xbmc.Player.__init__(self)
            self.estimateFinishTime = '00:00:00'
            self.realFinishTime = '00:00:00'
            self.itemDuration = '00:00:00'
            self.itemDBID = itemDBID
            self.itemType = itemType
            self.pomSlovnik = slovnik
            self.titulky = titulky
            # dummy call to fix weird error see:
            # http://bugs.python.org/issue7980
            try:
                datetime.strptime('2012-01-01', '%Y-%m-%d')
            except TypeError:
                datetime(*(time.strptime('2012-01-01', '%Y-%m-%d')[0:6]))
        except Exception:
            buggalo.onExceptionRaised({'self.itemDBID: ': self.itemDBID,
                                       'self.itemType: ': self.itemType})

    @staticmethod
    def executeJSON(request):
        # =================================================================
        # Execute JSON-RPC Command
        # Args:
        # request: Dictionary with JSON-RPC Commands
        # Found code in xbmc-addon-service-watchedlist
        # =================================================================
        rpccmd = json.dumps(request)  # create string from dict
        json_query = xbmc.executeJSONRPC(rpccmd)
        json_query = unicode(json_query, 'utf-8', errors='ignore')
        json_response = json.loads(json_query)
        return json_response

    @staticmethod
    def get_sec(time_str):
        # nasty bug appears only for 2nd and more attempts during session
        # workaround from: http://forum.kodi.tv/showthread.php?tid=112916
        try:
            t = datetime.strptime(time_str, "%H:%M:%S")
        except TypeError:
            t = datetime(*(time.strptime(time_str, "%H:%M:%S")[0:6]))
        return timedelta(hours=t.hour, minutes=t.minute, seconds=t.second)

    def setWatched(self):
        if self.itemType == u'episode':
            metaReq = {"jsonrpc": "2.0",
                       "method": "VideoLibrary.SetEpisodeDetails",
                       "params": {"episodeid": self.itemDBID,
                                  "playcount": 1},
                       "id": 1}
            self.executeJSON(metaReq)
        elif self.itemType == u'movie':
            metaReq = {"jsonrpc": "2.0",
                       "method": "VideoLibrary.SetMovieDetails",
                       "params": {"movieid": self.itemDBID,
                                  "playcount": 1},
                       "id": 1}
            self.executeJSON(metaReq)

    def createResumePoint(self, seconds, total):
        try:
            pomer = seconds / total
            if pomer < 0.05:
                return
            self.pomSlovnik.update({self.itemDBID: seconds})
        except Exception:
            buggalo.onExceptionRaised({'seconds: ': seconds})
        return

    def onPlayBackStarted(self):
        try:
            # ListItem.Duration je z databáze, bývá nepřesná v řádech minut
            # Player.TimeRemaining je přesnější
            if self.titulky is not None:
                self.setSubtitles(self.titulky[0])
            while True:
                xbmc.sleep(1000)
                self.itemDuration = xbmc.getInfoLabel(
                    'Player.TimeRemaining(hh:mm:ss)')
                if (self.itemDuration != '') and (self.itemDuration != '00:00:00'):
                    self.itemDuration = self.get_sec(self.itemDuration)
                    break
            # plánovaný čas dokončení 100 % přehrání
            self.estimateFinishTime = xbmc.getInfoLabel(
                'Player.FinishTime(hh:mm:ss)')
            if (not self.pomSlovnik) or (not self.itemDBID in self.pomSlovnik):
                return
            self.seekTime(self.pomSlovnik[self.itemDBID])
            del self.pomSlovnik[self.itemDBID]
        except Exception:
            buggalo.onExceptionRaised({'self.itemDuration: ': self.itemDuration,
                                       'self.estimateFinishTime: ': self.estimateFinishTime,
                                       'pomSlovnik: ': json.dumps(self.pomSlovnik, indent=2)})

    def onPlayBackEnded(self):
        self.setWatched()

    def onPlayBackStopped(self):
        try:
            # Player.TimeRemaining  - už zde nemá hodnotu
            # Player.FinishTime - kdy přehrávání skutečně zkončilo
            timeDifference = 55555
            timeRatio = 55555
            self.realFinishTime = xbmc.getInfoLabel(
                'Player.FinishTime(hh:mm:ss)')
            timeDifference = self.get_sec(self.estimateFinishTime) - \
                self.get_sec(self.realFinishTime)
            timeRatio = timeDifference.seconds / \
                float((self.itemDuration).seconds)
            # upravit podmínku na 0.05 tj. zbývá shlédnout 5%
            if abs(timeRatio) < 0.1:
                self.setWatched()
            else:
                self.createResumePoint((1 - timeRatio) * float((self.itemDuration).seconds),
                                       float((self.itemDuration).seconds))
        except Exception:
            buggalo.onExceptionRaised({'self.itemDuration: ': self.itemDuration,
                                       'self.estimateFinishTime: ': self.estimateFinishTime,
                                       'self.realFinishTime: ': self.realFinishTime,
                                       'timeDifference: ': timeDifference,
                                       'timeRatio: ': timeRatio, })

    def waitForChange(self):
        xbmc.sleep(2000)
        while True:
            pom = xbmc.getInfoLabel('Player.FinishTime(hh:mm:ss)')
            if pom != self.estimateFinishTime:
                self.estimateFinishTime = pom
                break
            xbmc.sleep(100)

    def onPlayBackResumed(self):
        self.waitForChange()

    def onPlayBackSpeedChanged(self, speed):
        self.waitForChange()

    def onPlayBackSeek(self, time, seekOffset):
        self.waitForChange()


resources/lib/translatedStrings.py
import xbmcaddon

__scriptid__ = 'plugin.video.sosac.ph'
__addon__ = xbmcaddon.Addon(id=__scriptid__)
ADDON_SETTINGS_GET = xbmcaddon.Addon(id=__scriptid__).getSetting
ADDON_SETTINGS_SET = xbmcaddon.Addon(id=__scriptid__).setSetting

CSFD_BASE = 'http://www.csfd.cz/'
ZEBRICKY_FILMY_NEJ = 'zebricky/nejlepsi-filmy/?show=complete'
ZEBRICKY_SPEC = 'zebricky/specificky-vyber/'
ZEBRICKY_FILMY_SPEC = (ZEBRICKY_SPEC +
                       '?type=0&origin=&genre=&year_from=&year_to=&actor=&director=&ok=Zobrazit&_form_=charts')
ZEBRICKY_TVSHOW_NEJ = 'zebricky/nejlepsi-serialy/?show=complete'
ZEBRICKY_TVSHOW_SPEC = (ZEBRICKY_SPEC +
                        '?type=3&origin=&genre=&year_from=&year_to=&actor=&director=&ok=Zobrazit&_form_=charts')
ZEBRICKY_FILMY_SPEC_GENRE = 'type=0'
ZEBRICKY_TVSHOW_SPEC_GENRE = 'type=3'
OCENENI_OSCAR = 'oceneni/1-oscars/'
OCENENI_OSCAR_ROKY = '?years='
OCENENI_ZLATA_PALMA = 'oceneni/4-cannes-iff/'
OCENENI_ZLATA_PALMA_ROKY = '?years='
# strings used in sosac.py nad sutils.py
LIBRARY_TYPE_VIDEO = "video"
LIBRARY_TYPE_TVSHOW = "tvshow"
# playing from restored position
POKRACOVAT = __addon__.getLocalizedString(30208)
OD_ZACATKU = __addon__.getLocalizedString(30209)
OD_MINULE_POZICE = __addon__.getLocalizedString(30210)
# main menu items
MOVIES = __addon__.getLocalizedString(30300)
TV_SHOWS = __addon__.getLocalizedString(30301)
MOVIES_BY_GENRES = __addon__.getLocalizedString(30302)
MOVIES_MOST_POPULAR = __addon__.getLocalizedString(30303)
TV_SHOWS_MOST_POPULAR = __addon__.getLocalizedString(30304)
MOVIES_RECENTLY_ADDED = __addon__.getLocalizedString(30305)
TV_SHOWS_RECENTLY_ADDED = __addon__.getLocalizedString(30306)
SPRAVCE_ODBERU = __addon__.getLocalizedString(30310)
MOVIES_BY_YEAR = __addon__.getLocalizedString(30311)
CSFD_MAIN = __addon__.getLocalizedString(30323)
CSFD_LADDERS = __addon__.getLocalizedString(30317)
CSFD_AWARDS = __addon__.getLocalizedString(30318)
BEST_MOVIES = __addon__.getLocalizedString(30319)
BEST_MOVIES_BY_GENRE = __addon__.getLocalizedString(30320)
BEST_TV_SHOWS = __addon__.getLocalizedString(30321)
BEST_TV_SHOWS_BY_GENRE = __addon__.getLocalizedString(30322)
OSCARS = __addon__.getLocalizedString(30324)
GOLDEN_PALM = __addon__.getLocalizedString(30325)
# context menu items
ADD_ALL_TO_LIBRARY = __addon__.getLocalizedString(30307)
ADD_TO_LIBRARY = __addon__.getLocalizedString(30308)
REMOVE_FROM_SUBSCRIPTION = __addon__.getLocalizedString(30309)
SUBSCRIBE = __addon__.getLocalizedString(30312)
REMOVE_ALL_FROM_SUBSCRIPTION = __addon__.getLocalizedString(30313)
SUBSCRIBE_ALL_TV_SHOWS = __addon__.getLocalizedString(30314)
REFRESH_SUBSCRIPTION = __addon__.getLocalizedString(30217)


default.py
# -*- coding: UTF-8 -*-
# /*
#  *      Copyright (C) 2013 Libor Zoubek + jondas
#  *
#  *
#  *  This Program is free software; you can redistribute it and/or modify
#  *  it under the terms of the GNU General Public License as published by
#  *  the Free Software Foundation; either version 2, or (at your option)
#  *  any later version.
#  *
#  *  This Program is distributed in the hope that it will be useful,
#  *  but WITHOUT ANY WARRANTY; without even the implied warranty of
#  *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
#  *  GNU General Public License for more details.
#  *
#  *  You should have received a copy of the GNU General Public License
#  *  along with this program; see the file COPYING.  If not, write to
#  *  the Free Software Foundation, 675 Mass Ave, Cambridge, MA 02139, USA.
#  *  http://www.gnu.org/copyleft/gpl.html
#  *
#  */

import sys
import xbmcaddon
import xbmcutil
import util
from resources.lib.sosac import SosacContentProvider
from resources.lib.sutils import XBMCSosac

__scriptid__ = 'plugin.video.sosac.ph'
__scriptname__ = 'sosac.ph'
__addon__ = xbmcaddon.Addon(id=__scriptid__)
__language__ = __addon__.getLocalizedString
__set__ = __addon__.getSetting

settings = {'downloads': __set__('downloads'),
            'quality': __set__('quality'),
            'subs': __set__('subs') == 'true',
            'add_subscribe': __set__('add_subscribe'),
            'force-ch': __set__('force-ch') == 'true',
            'force-sort': __set__('force-sort')}


reverse_eps = __set__('order-episodes') == '0'
force_english = __set__('force-english') == 'true'
use_memory_cache = __set__('use-memory-cache') == 'true'

util.info("URL: " + sys.argv[2])
params = util.params()
if params == {}:
    xbmcutil.init_usage_reporting(__scriptid__)

util.info("Running sosac provider with params: " + str(params))
XBMCSosac(SosacContentProvider(reverse_eps=reverse_eps, force_english=force_english,
                               use_memory_cache=use_memory_cache), settings,
          __addon__).run(params)
